apps/emqx/src/emqx_channel.erl

```diff
@@ -1075,6 +1070,10 @@ return_connack(AckPacket, Channel) ->
             },
             {Packets, NChannel2} = do_deliver(Publishes, NChannel1),
             Outgoing = [?REPLY_OUTGOING(Packets) || length(Packets) > 0],
+            % NOTE
+            % Session timers are not restored here, so there's a tiny chance that
+            % the session becomes stuck, when it already has no place to track new
+            % messages.
```

Comment:
No, the opposite actually, it's the consequence of it. I just wanted to highlight it, and leave the behavior unchanged. Do you think it would be better to just restore the timers and drop the comment?

Reply:
No, I think we should leave that to another PR.  I thought this was an intermediate commit leftover, sorry.